core/math/audio_frame.h

```diff
@@ -63,60 +63,81 @@ struct AudioFrame {
 			float r;
 		};
 #endif
-		float levels[2] = { 0.0 };
+		float levels[2] = { 0, 0 };
 	};
 
-	_ALWAYS_INLINE_ const float &operator[](int p_idx) const {
-		DEV_ASSERT((unsigned int)p_idx < 2);
-		return levels[p_idx];
-	}
-	_ALWAYS_INLINE_ float &operator[](int p_idx) {
-		DEV_ASSERT((unsigned int)p_idx < 2);
-		return levels[p_idx];
-	}
-
-	_ALWAYS_INLINE_ AudioFrame operator+(const AudioFrame &p_frame) const { return AudioFrame(left + p_frame.left, right + p_frame.right); }
-	_ALWAYS_INLINE_ AudioFrame operator-(const AudioFrame &p_frame) const { return AudioFrame(left - p_frame.left, right - p_frame.right); }
-	_ALWAYS_INLINE_ AudioFrame operator*(const AudioFrame &p_frame) const { return AudioFrame(left * p_frame.left, right * p_frame.right); }
-	_ALWAYS_INLINE_ AudioFrame operator/(const AudioFrame &p_frame) const { return AudioFrame(left / p_frame.left, right / p_frame.right); }
-
-	_ALWAYS_INLINE_ AudioFrame operator+(float p_sample) const { return AudioFrame(left + p_sample, right + p_sample); }
-	_ALWAYS_INLINE_ AudioFrame operator-(float p_sample) const { return AudioFrame(left - p_sample, right - p_sample); }
-	_ALWAYS_INLINE_ AudioFrame operator*(float p_sample) const { return AudioFrame(left * p_sample, right * p_sample); }
-	_ALWAYS_INLINE_ AudioFrame operator/(float p_sample) const { return AudioFrame(left / p_sample, right / p_sample); }
-
-	_ALWAYS_INLINE_ void operator+=(const AudioFrame &p_frame) {
+	constexpr const float &operator[](size_t p_idx) const {
+		switch (p_idx) {
+			case 0:
+				return left;
+			case 1:
+				return right;
+			default:
+				return levels[p_idx];
```

Comment:
This should be an error instead IMO, the dev assert is there to make the error explicit and reliable, now it becomes a contextual segfault

Reply:
Those error macros don't play nice with `constexpr` environments out of the gate, but an early catch would still be nice in some form. Can't use `consteval` nor `std::is_constant_evaluated` in C++17, but `__builtin_is_constant_evaluated` *should* do the trick.